core/templates/dev/head/AppSpec.ts

```diff
@@ -16,8 +16,18 @@
  * @fileoverview Unit tests for generic services.
  */
 
+import { RuleObjectFactory } from 'domain/exploration/RuleObjectFactory.ts';
```

Comment:
I'm not sure I understand why these need to be added here. They don't seem to be used in the test?

There should probably be documentation somewhere to explain -- either here if it's specific to this, or in our wiki if it's a general principle to follow. In any case I think it would be nice to add a comment above line 25 explaining why lines 26-29 are needed. Do the same in all other similar files.

Reply:
This is not specific to this file, this is the case in many files. What's happening is that since we do not use isolated tests, these spec files depended upon the upgraded services and therefore angular mock could not interpret them. 
This would be more suited to the wiki than a comment. I'll add this to the Angular wiki.

Reply:
Is this something that would be removed after the full migration to Angular 8?

If so, we should add a TODO to remove them in the future, like Vojta did for some of the webpack requires. I realize that that might be repetitive, but we should do that anyway to make clear that this isn't a long-term practice to be followed.

Reply:
The statements can be removed once the corresponding non-spec file is upgraded.
Added the TODOs!